### PR TITLE
Support vagrant lookup path

### PR DESCRIPTION
**Pull request by JoProvost**
This will look in parent folders if the provided folder does not contain a Vagrantfile.
See https://www.vagrantup.com/docs/vagrantfile/

Also support the following env variables:
 - VAGRANT_DOTFILE_PATH
 - VAGRANT_VAGRANTFILE
 - VAGRANT_CWD

See https://www.vagrantup.com/docs/other/environmental-variables.html

### CODE REVIEW

File: tests/test_vagrant_source.py [hunk 1]
```diff
@@ -19,6 +19,53 @@
...
 
 
+class TestFindVagrantRoot(unittest.TestCase):
+    def setUp(self):
+        self.origin_vagrant_dotfile_path = vagrant_source.VAGRANT_DOTFILE_PATH
+        self.origin_vagrant_vagrantfile = vagrant_source.VAGRANT_VAGRANTFILE
+        self.origin_vagrant_cwd = vagrant_source.VAGRANT_CWD
+        self.addCleanup(self.cleanupVagrantVariables)
+
+        self.valid_vagrant_root = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'vagrant_project')
```
Comment (lindycoder) [inline tests/test_vagrant_source.py]: maybe make "os.path.join(os.path.dirname(os.path.abspath(__file__)), ????)" a function, it appears at least 6 times in this file only

File: tests/test_vagrant_source.py [hunk 2]
```diff
@@ -19,6 +19,53 @@
...
+        vagrant_source.VAGRANT_VAGRANTFILE = self.origin_vagrant_vagrantfile
+        vagrant_source.VAGRANT_CWD = self.origin_vagrant_cwd
+
+    def test_default_values(self):
+        self.assertEqual(vagrant_source.VAGRANT_DOTFILE_PATH, '.vagrant')
+        self.assertEqual(vagrant_source.VAGRANT_VAGRANTFILE, 'Vagrantfile')
+        self.assertEqual(vagrant_source.VAGRANT_CWD, os.getcwd())
+
+    def test_on_a_sub_folder_of_a_valid_vagrant_root_path(self):
+        vagrant_root = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'vagrant_project')
```
Comment (lindycoder) [inline tests/test_vagrant_source.py]: could use self.valid_vagrant_root

File: tests/test_vagrant_source.py (hunk 1 above)
Comment (JoProvost) [inline tests/test_vagrant_source.py]: Fixed, thank you!

File: tests/test_vagrant_source.py (hunk 2 above)
Comment (JoProvost) [inline tests/test_vagrant_source.py]: Fixed, thank you!

**Comment by lindycoder** [comment]
👍

**Pull request merged**